### PR TITLE
feat: autocomplete computed member expression

### PR DESCRIPTION
**Pull request by TrySound**
Fixes https://github.com/webstudio-is/webstudio/issues/2790 https://discord.com/channels/955905230107738152/1203115803910873218

Here fixed the issue with completing array indexes or object keys unsupported with dot syntax.

Screenshot 2024-02-04 at 17 34 37
Screenshot 2024-02-04 at 17 33 20
Screenshot 2024-02-04 at 17 33 27

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

**Comment by kof** [comment]
OOOO AMAZING

**Comment by kof** [comment]
Is it possible to sort the array by number?

image

**Comment by kof** [comment]
more feedback https://share.descript.com/view/P4s5EkJUnx5

**Comment by TrySound** [comment]
Please show what your editor does when autocomplete without "."?

**Review by kof — APPROVED** [review]

File: apps/builder/app/builder/shared/expression-editor.tsx [hunk 1]
```diff
@@ -80,13 +85,95 @@ const VariablesData = Facet.define<{
...
+
+/// Helper function for defining JavaScript completion sources. It
+/// returns the completable name and object path for a completion
+/// context, or undefined if no name/property completion should happen at
+/// that position. For example, when completing after `a.b.c` it will
+/// return `{path: ["a", "b"], name: "c"}`. When completing after `x`
+/// it will return `{path: [], name: "x"}`. When not in a property or
+/// name, it will return undefined if `context.explicit` is false, and
+/// `{path: [], name: ""}` otherwise.
+const completionPath = (
```
Comment (kof) [inline apps/builder/app/builder/shared/expression-editor.tsx:139]: lets add some tests for this?

Comment (TrySound) [reply apps/builder/app/builder/shared/expression-editor.tsx:139]: Have no idea how to test this 😄

Comment (kof) [reply apps/builder/app/builder/shared/expression-editor.tsx:139]: isn't there a way to get that object from codemirror? check how they test it? Maybe its not too hard to just take a snapshot of it from from the browser?

Comment (TrySound) [reply apps/builder/app/builder/shared/expression-editor.tsx:139]: I would prefer to spend time on porting it to original package

**Pull request merged**